### PR TITLE
Expose BriteDatabase#getWriteableDatabase

### LINKED ISSUE #159: BriteDatabase should expose getWriteableDatabase for use with SQLDelight
Currently you have to pass your instance of `SqlLiteOpenHelper` around alongside your `BriteDatabase` in order to use SQLDelight compiled statements with sqlbrite. It would be much nicer to just pass the `BriteDatabase` instance and be able to retrieve the database via `BriteDatabase#getWriteableDatabase`.

### PR DESCRIPTION
**Pull request by burntcookie90**
Expose the BriteDatabase's helper db via getWriteableDatabase.

Fixes #159

### CODE REVIEW

**Comment by burntcookie90** [comment]
I thought about exposing `getReadableDatabase` also, since it's right there and maybe it could be useful? Or just leave it be.

**Comment by JakeWharton** [comment]
I don't see a reason not to also expose readable.

File: sqlbrite/src/main/java/com/squareup/sqlbrite/BriteDatabase.java [hunk 1]
```diff
@@ -149,8 +149,7 @@ SQLiteDatabase getReadableDatabase() {
     return db;
   }
 
-  // Package-private to avoid synthetic accessor method for 'transaction' instance.
```
Comment (AlecStrong) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BriteDatabase.java]: can you include the comments for `getReadableDatabase/getWritableDatabase` off `SQLiteDatabase` if there are any similar to what we do for the other copied apis

File: sqlbrite/src/main/java/com/squareup/sqlbrite/BriteDatabase.java [hunk 2]
```diff
@@ -135,7 +135,25 @@ public void setLoggingEnabled(boolean enabled) {
...
+   * <p class="caution">Like {@link SQLiteOpenHelper#getWritableDatabase}, this method may
+   * take a long time to return, so you should not call it from the
+   * application main thread, including from
+   * {@link android.content.ContentProvider#onCreate ContentProvider.onCreate()}.
+   *
+   * @throws android.database.sqlite.SQLiteException if the database cannot be opened
+   * @return a database object valid until {@link SQLiteOpenHelper#getWritableDatabase}
+   *     or {@link #close} is called.
+   */
+  public SQLiteDatabase getReadableDatabase() {
```
Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BriteDatabase.java]: Can you add `@NonNull @CheckResult @WorkerThread` to these two methods as well?

**Comment by JakeWharton** [comment]
Ok I have one last concern: we're exposing a cached instance that is never closed (and never should be) to the user. The documentation of this method explicitly says that you need to close the returned instance when you're done with it, but doing so will poison the instance we're using internally. I think if we're going to expose these, we need to make them pass-through factories to the underlying database and change the existing methods which cache the instance to be for internal use only.

**Comment by AlecStrong** [comment]
What do you mean by "pass-through factories to the underlying database"

If you still get a `SQLiteDatabase` back at the end I'm okay with that

**Comment by JakeWharton** [comment]
```java
public SQliteDatabase getWriteableDatabase() {
  return helper.getWritableDatabase();
}
```

**Comment by burntcookie90** [comment]
That was my original (having not looked at the source) plan

**Comment by JakeWharton** [comment]
Yes let's do that instead. So rename the existing methods that do caching
to something else and then expose these pass-through versions publicly.

**Comment by AlecStrong** [comment]
That still suffers from the same problem though... `SQLiteOpenHelper` does it's own caching of a `SQLiteDatabase` instance and so closing the returned database will close our cached one.

We probably shouldn't have our own cached version at all - since `SQLiteOpenHelper` does it's own caching as long as the user never closes a `SQLiteDatabase` given to them the behavior is exactly the same, it uses a single cached version.

However, if the user does close a database, by calling `getWritableDatabase` or `getReadableDatabase` we're protecting ourselves

**Comment by JakeWharton** [comment]
Got it. That sounds good.

**Comment by burntcookie90** [comment]
Okie dokes, I'll take a whack at it this evening.

**Comment by AlecStrong** [comment]
Siiiiiiiiiiick looks good :+1:

**Comment by JakeWharton** [comment]
I'll release later tonight or tomorrow.

**Comment by burntcookie90** [comment]
Cool thanks

**Pull request merged**